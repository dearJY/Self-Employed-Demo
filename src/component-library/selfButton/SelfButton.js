import * as React from 'react';
import styled from 'styled-components';

const Button = styled.button`
    background-color:rgb(44, 160, 28);
    border-radius: 3px;
    color:white;
    border: none;
    padding:10px;
    font-size:15px;
    cursor:pointer;
`;

export const SelfButton = (props) => {
    return <Button onClick={props.onClick}>{props.text}</Button>
};
