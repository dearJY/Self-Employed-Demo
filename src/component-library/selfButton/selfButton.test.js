import React from 'react';
import {SelfButton} from "./SelfButton";
import {shallow} from 'enzyme';

test("renders selfButton correctly", () => {
    const tree = shallow(
        <SelfButton text="click me"></SelfButton>
    );
    expect(tree).toMatchSnapshot();
});