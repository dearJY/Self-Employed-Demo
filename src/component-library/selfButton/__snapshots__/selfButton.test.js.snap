// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders selfButton correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelfButton
    text="click me"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "click me",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": "click me",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    background-color:rgb(44, 160, 28);
    border-radius: 3px;
    color:white;
    border: none;
    padding:10px;
    font-size:15px;
    cursor:pointer;
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "click me",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "click me",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    background-color:rgb(44, 160, 28);
    border-radius: 3px;
    color:white;
    border: none;
    padding:10px;
    font-size:15px;
    cursor:pointer;
",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
